core/chains/evm/config/chain_specific_config.go

```diff
@@ -325,6 +325,20 @@ func setChainSpecificConfigDefaultSets() {
 	optimismGoerli := optimismKovan
 	optimismGoerli.linkContractAddress = "0xdc2CC710e42857672E7907CF474a69B63B93089f"
 
+	// Optimism's Bedrock upgrade uses EIP-1559.
+	optimismBedrock := fallbackDefaultSet
+	optimismBedrock.eip1559DynamicFees = true
+	optimismBedrock.blockEmissionIdleWarningThreshold = 30 * time.Second
+	optimismBedrock.nodeDeadAfterNoNewHeadersThreshold = 60 * time.Second
+	optimismBedrock.blockHistoryEstimatorBlockHistorySize = 24
+	optimismBedrock.chainType = config.ChainOptimismBedrock
+	optimismBedrock.ethTxResendAfterThreshold = 30 * time.Second
+	optimismBedrock.finalityDepth = 200				 // Although Bedrock is re-org resistant for 10 blocks on L1, L2 blocks are considered final when they are included in L2 final block (PoS).
```

Comment:
Comment doesn't help much here. Why was the value of 200 chosen?

Reply:
L1-PoS: “Every 32 slots(epoch) each validator on the network has the opportunity to vote in favor of the epoch. It takes two justified epochs (a “justified” epoch meaning the majority of validators agreed on it) for those epochs, and all the blocks inside of them, to be considered finalized.”

32 L1 blocks (best case scenario) X
6 L2 blocks (2s block time) =
192 blocks - if the majority voted for the same checkpoints.

I'll leave a comment explaining the above.

Reply:
hmm safer to wait for L1 finality i.e. 64 blocks no? Also  we should update the mainnet to 64 as well

Reply:
There is no specific number that guarantees finality. This only depends on the voting outcome of the validators. This could be anything, although the protocol has to have some sort of enforcement eventually. From a quick experiment in Mainnet the difference between latest and last finalized block was up to 92 blocks difference. To properly account for that we need to implement a new confirmer that listens for safe or finalized blocks (depending on our security assumptions) and confirms finalization

Reply:
Yeah we deliberately have not taken into account L1 security on our L2 implementations. Doing so is complex and has been in the backlog for a long time.

Reply:
Yeah, that is correct. > 192 will be potentially safe rather than finalized.